arches/app/models/graph.py

```diff
@@ -2828,99 +2593,103 @@ def restore_state_from_serialized_graph(self, serialized_graph):
         Restores a Graph's state from a serialized graph, and creates a
         new editable_future_graph
         """
-        models.NodeGroup.objects.filter(
-            pk__in=[
-                nodegroup.pk
-                for nodegroup in self.get_nodegroups(force_recalculation=True)
-            ]
-        ).delete()
-        models.Node.objects.filter(
-            pk__in=[node.pk for node in self.nodes.values()]
-        ).delete()
-        models.Edge.objects.filter(
-            pk__in=[edge.pk for edge in self.edges.values()]
-        ).delete()
-        models.CardModel.objects.filter(
-            pk__in=[card.pk for card in self.cards.values()]
-        ).delete()
-        models.CardXNodeXWidget.objects.filter(
-            pk__in=[
-                card_x_node_x_widget.pk
-                for card_x_node_x_widget in self.widgets.values()
-            ]
-        ).delete()
+        with transaction.atomic():
+            editable_future_graph = Graph.objects.filter(
+                source_identifier_id=self.pk
+            ).first()
+            if editable_future_graph:
+                editable_future_graph.delete()
 
-        for serialized_nodegroup in serialized_graph["nodegroups"]:
-            for key, value in serialized_nodegroup.items():
-                try:
-                    serialized_nodegroup[key] = uuid.UUID(value)
-                except:
-                    pass
+            # ensures any resources that were related to the source graph are not deleted
+            self.pk = uuid.uuid4()
+            self.delete()
```

Comment:
but we also want to delete the graph. I'm unsure decomposing this will do anything for

```
graph = models.ForeignKey(GraphModel, db_column="graphid", on_delete=models.CASCADE)
```

on ResourceInstance.

Reply:
Well from a code read, we're not deleting the graph, and I just checked the queries to be sure. We set the PK to a random value, so that super().delete() will shoot a bunch of blanks, firing off deletes on a bunch of tables that will never match on anything. Since we're not deleting the graph, I was thinking it would be clearer to decompose that out and avoid shooting the blanks:

```py
BEFORE:  a6cd3760-199b-4071-bd15-307fb04e53d2
AFTER:  de19d77d-4b59-41ae-bdee-aa7669cda961
```
Notice the deletes are for the random uuid (also why I was mentioning the very very rare risk of collision):
```
-> super(Graph, self).delete()
(Pdb) print(connection.queries)
[{'sql': 'SELECT "cards"."cardid" FROM "cards" WHERE "cards"."graphid" IN (\'de19d77d-4b59-41ae-bdee-aa7669cda961\'::uuid)', 'time': '0.003'}, {'sql': 'SELECT "edges"."edgeid" FROM "edges" WHERE "edges"."graphid" IN (\'de19d77d-4b59-41ae-bdee-aa7669cda961\'::uuid)', 'time': '0.002'}, {'sql': 'SELECT "graphs"."graphid" FROM "graphs" WHERE "graphs"."source_identifier" IN (\'de19d77d-4b59-41ae-bdee-aa7669cda961\'::uuid)', 'time': '0.001'}, {'sql': 'SELECT "graphs_x_published_graphs"."publicationid" FROM "graphs_x_published_graphs" WHERE "graphs_x_published_graphs"."graphid" IN (\'de19d77d-4b59-41ae-bdee-aa7669cda961\'::uuid)', 'time': '0.002'}, {'sql': 'SELECT "nodes"."nodeid" FROM "nodes" WHERE "nodes"."graphid" IN (\'de19d77d-4b59-41ae-bdee-aa7669cda961\'::uuid)', 'time': '0.001'}, {'sql': 'SELECT "resource_instances"."resourceinstanceid" FROM "resource_instances" WHERE "resource_instances"."graphid" IN (\'de19d77d-4b59-41ae-bdee-aa7669cda961\'::uuid)', 'time': '0.001'}, {'sql': 'DELETE FROM "functions_x_graphs" WHERE "functions_x_graphs"."graphid" IN (\'de19d77d-4b59-41ae-bdee-aa7669cda961\'::uuid)', 'time': '0.001'}, {'sql': 'DELETE FROM "resource_x_resource" WHERE ("resource_x_resource"."resourceinstancefrom_graphid" IN (\'de19d77d-4b59-41ae-bdee-aa7669cda961\'::uuid) OR "resource_x_resource"."resourceinstanceto_graphid" IN (\'de19d77d-4b59-41ae-bdee-aa7669cda961\'::uuid))', 'time': '0.001'}, {'sql': 'DELETE FROM "graphs_x_mapping_file" WHERE "graphs_x_mapping_file"."graphid" IN (\'de19d77d-4b59-41ae-bdee-aa7669cda961\'::uuid)', 'time': '0.001'}, {'sql': 'DELETE FROM "graphs" WHERE "graphs"."graphid" IN (\'de19d77d-4b59-41ae-bdee-aa7669cda961\'::uuid)', 'time': '0.001'}]
```

Reply:
I guess I'm talking about:

```
{'sql': 'DELETE FROM "graphs" WHERE "graphs"."graphid" IN (\'de19d77d-4b59-41ae-bdee-aa7669cda961\'::uuid)', 'time': '0.001'}
```

so that the current graph object gets deleted. I know the `Graph.delete` method manually handles cards, nodes, widgets, nodegroups, and edges in a way that wouldn't be affected by changing the graphid. `graph_x_published_graph` is interesting, I haven't checked that those are also deleted, but I will.

I'm not against the decomposition here, but at some point the graph itself needs to be deleted, so not I'm entirely sure that the decomposition will handle what we need it to.

Reply:
But `de19d77d-4b59-41ae-bdee-aa7669cda961` isn't the current object, it's the random one, right?

BEFORE is self.pk, then
AFTER is after setting self.pk = uuid.uuid4()
```
BEFORE:  a6cd3760-199b-4071-bd15-307fb04e53d2
AFTER:  de19d77d-4b59-41ae-bdee-aa7669cda961
```

Reply:
ah. yeah I get it 😞 . Def something to handle for